pals/core.py

```diff
@@ -84,7 +84,7 @@ def lock(self, name, **kwargs):
         name = self._lock_name(name)
         kwargs.setdefault('blocking', self.blocking_default)
         kwargs.setdefault('acquire_timeout', self.acquire_timeout_default)
-        return Lock(self.engine.connect(), lock_num, name, **kwargs)
+        return Lock(self.engine, lock_num, name, **kwargs)
```

Comment:
Lock expects an engine, but a connection was previously being passed. That happened to work in pre-2.0.